libcudacxx/include/cuda/__memory_resource/resource.h

```diff
@@ -99,10 +100,29 @@ template <class _Resource, class... _Properties>
 _LIBCUDACXX_CONCEPT async_resource_with =
   async_resource<_Resource> && _CUDA_VSTD::__all_of<has_property<_Resource, _Properties>...>;
 
+template <bool _Convertible>
+struct __different_resource__
+{
+  template <class _OtherResource>
+  static constexpr bool __value(_OtherResource*) noexcept
+  {
+    return resource<_OtherResource>;
+  }
```

Comment:
Remark: I assume you want to be C++11 compatible, otherwise this could be a variable template.